### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by Evangelink**

### CODE REVIEW

File: docs/contributing-analyzer.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
+# Building, Testing and Debugging the Analyzer
+
+All C# and VB.NET analyzers present in SonarLint for Visual Studio, in the SonarQube SonarC# Plugin or in the
+SonarQube SonarVB plugin are being developed here.
+These analyzers rely on Roslyn 1.0 API.
+
+## Working with the code
```
Comment (janos-ss) [inline docs/contributing-analyzer.md]: I recommend to consistently add a blank line after section headings. It makes it easier to read as plain text (which is one perfectly reasonable way to consume markdown)

File: docs/contributing-analyzer.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+### Unit Tests
+You can run the Unit Tests via Test Explorer of Visual Studio.
+
+### Integration Tests
+If this is the first time you are running the ITs you will need to run the following command:
+git submodule update --init --recursive
+
+To run the ITs you will need to follow this pattern:
+1. Open the `Developer Command Prompt for VS2015`
```
Comment (janos-ss) [inline docs/contributing-analyzer.md]: Similar to blank lines before section headings, I also recommend to put a blank line between paragraphs and lists

File: docs/contributing-analyzer.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+2. Hit `F5` to launch the experimental instance of Visual Studio
+3. Open one of the following solutions from the experimental instance:
+  * [Akka.NET](akka.net/src/Akka.sln)
+  * [Nancy](Nancy/src/Nancy.sln)
+  * [Ember-MM](Ember-MM/Ember Media Manager.sln)
+4. Turn on your new rule in [Validation Ruleset](ValidationRuleset.ruleset), review the results, improve, and setup the regression test once you are satisfied.
+Note: the solutions have been pre-configured to use this ruleset on all their projects.
+
+## Contributing
+Please see [Contributing Code](https://github.com/SonarSource/sonar-csharp/blob/master/CONTRIBUTING.md) for details on
```
Comment (janos-ss) [inline docs/contributing-analyzer.md]: I could be wrong, but I think simple relative paths would work

**Comment by Evangelink** [comment]
Rebase on master done

**Pull request merged**